simulator/mips/mips_instr.cpp

```diff
@@ -216,12 +226,27 @@ FuncInstr::FuncInstr( uint32 bytes, Addr PC,
 void FuncInstr::initFormat()
 {
     bool is_R = ( instr.asR.opcode == 0x0);
+    bool is_RI = ( instr.asR.opcode == 0x1);
     uint8 ident = is_R ? instr.asR.funct : instr.asR.opcode;
 
-    for ( const auto& entry : isaTable)
+    if ( is_R)
+    {    
+        const auto& entry = isaMapR.at( ident);
```

Comment:
Please avoid code duplication, use reference to the correct `std::unordered_map`